### DISCUSSION TITLE
Cant use local png in diagram

### DISCUSSION THREAD

**Discussion by cwilliamson-bti** [plantuml/plantuml#1792] [Q&A] [answered]
Hi!
I'm trying to use a local PNG in a diagram but keep getting the message `"(Cannot decode)"` in the object. Below is the code for what I'm trying to do:
```plantuml
@startuml
object "<img:led.png>" as led
led -> A : Hello
@enduml
```

`led.png` is located in the same folder as the `plantuml.jar` file which is being ran through doxygen.
- Is this the correct way to be loading a local PNG to be used?

**Comment by The-Lum** [comment]
Hi @cwilliamson-bti,

- Could you put your `led.png` in the same folder as the original source file containing the `@startuml`/`@enduml`?

Regards,
Th.

**Reply by cwilliamson-bti** [reply]
Hi!

Sorry for the delayed response. I tried that as well and still was unable to get it to work. Still getting this output:
![image](https://github.com/plantuml/plantuml/assets/144272504/83269020-0b5f-400a-ac40-fd9640084b35)

It does seem to be an issue with finding the file since if I change the name to something random it fails in the same way, I'm not sure where else I can store it though as I've copied it to all locations I can think it would look.

Cheers

**Reply by The-Lum** [reply]
Hi @cwilliamson-bti,

- How do you run `plantuml`?
- What is your `plantuml` version?
- Where is your `source_file.puml`?
- Where is your `led.png`?

Regards,
Th.

**Reply by cwilliamson-bti** [reply]
Hi,


    How do you run plantuml?
        From Doxygen
    What is your plantuml version?
        v1.2024.5
    Where is your source_file.puml?
        I think this would be the doxygen page I use plantuml diagrams in, it's in the same directory as the led.png
    Where is your led.png?
        In the same directory as the source file.

Thanks!

**Reply by The-Lum** [reply]
Hi,

- Perhaps ask question directly on `Doxygen` forum?
- https://github.com/doxygen/doxygen/discussions

See perhaps this configuration:
- https://stackoverflow.com/questions/23685689/how-to-include-iuml-path-to-generate-plantuml-diagram-in-doxygen


_[FYI @albert-github]_
Regards,
Th.

**Reply by The-Lum** [reply]
See perhaps also:
- https://github.com/doxygen/doxygen/issues/5921

And:
> Problem is that in the doxygen configuration file the PLANTUML_INCLUDE_PATH
isn't set. Plantuml looks in that case in output directory.
>
>Setting "PLANTUML_INCLUDE_PATH = ." solves this problem.
>
>I think this problem is not a bug but usage.

**Reply by albert-github** [reply, marked as answer]
A small test with doxygen shows that `PLANTUML_INCLUDE_PATH = .` is the way to go.

@cwilliamson-bti  for the record and in case it is not working for you I used:
- doxygen version 1.11.0 
- plantuml 1.2024.5beta4
- dot version 2.44.1

  Which versions are you using?

**Reply by cwilliamson-bti** [reply]
Thank you that was the issue!
